ionoscloud/resource_cdn_distribution.go

```diff
@@ -91,6 +91,11 @@ func resourceCDNDistribution() *schema.Resource {
 										Description: "Enable or disable WAF to protect the upstream host.",
 										Required:    true,
 									},
+									"sni_mode": {
+										Type:        schema.TypeString,
+										Description: "The SNI (Server Name Indication) mode of the upstream host. It supports two modes: 'distribution' and 'origin', for more information about these modes please check the resource docs.",
+										Required:    true,
+									},
```

Comment:
should we have here a validation for the "origin" and "distribution" values?

Reply:
@digna-ionos no, in the beginning I added (for other resources) all kinds of validations but it turned out that they are harder to maintain since all the changes in the API related to validation need to be reflected in Terraform, so now we let the API do all validation-related work.